### ISSUE TITLE
[BUG] Date & Time Event Filter does not work with Events Search [ver. 5.0.52 dev]

### ISSUE THREAD

**Issue by knguyenjahiel** [rokwire/illinois-app#3633]
**Describe the bug**
Date & Time Event Filter does not work with Events Search in the following ways: 
1. Results show events outside of the selected date and time 
2. The initial total number of relevant events increases as I scroll to include events past my selected timeframe.

**To Reproduce**
1. Events Feed >> Filter
4. Date & Time: Next Month (09/01-09/30)
6. Search "Meditation"
7. Results show events from Aug 26 to Dec 29 
8. Total number of relevant events increases as you scroll

**Expected behavior**
Results from Event Searches should be based on the filters set by the user.

**Comment by mihail-varbanov** [comment]
Search panel was designed to only search for text without any filtering or sorting capabilities. However I see the point of having them there. The problem is that this brings the necessity of ability to update the filters or sort order directly from the search panel. It also affects the Map panel that should also be able to search & filter events at the same time. Anyway, I think I did this all.

Search panel now appears as a kind of duplication of events home panel that also adds search capabilities. Maybe for the long run we should think on merging them, i.e. tapping on search in the events home panel to show search controls, but we should carefully design it as the UX may get too complex and confusing.

Regarding #5, this was a bug that I fixed. Thanks for pointing this out.

This all will get available in version 5.0.54.
